Stream_support/include/CGAL/IO/PLY/PLY_writer.h

```diff
@@ -298,6 +298,42 @@ class Char_property_printer
   }
 };
 
+
+template <typename Index,
+          typename PropertyMap,
+          typename VectorType = typename boost::property_traits<PropertyMap>::value_type,
+          typename ElementType = typename VectorType::value_type>
+class Simple_property_vector_printer
+  : public Abstract_property_printer<Index>
+{
+  PropertyMap m_pmap;
+public:
+  Simple_property_vector_printer(const PropertyMap& pmap) : m_pmap(pmap) { }
+
+  virtual void print(std::ostream& stream, const Index& index)
+  {
+    const VectorType& vec = get(m_pmap, index);
+    if(get_mode(stream) == CGAL::IO::ASCII)
+    {
+      stream << vec.size();
+      for(const ElementType& v : vec)
+      {
+        stream << " " << v;
+      }
+    }
+    else
+    {
+      unsigned char size = (unsigned char)(vec.size());
```

Comment:
I don't like this potentially-overflowing cast.

I copied it from existing code that uses the same logic:

https://github.com/CGAL/cgal/blob/acbd1ee182211c6309f355925927802f036fbe13/Surface_mesh/include/CGAL/Surface_mesh/IO/PLY.h#L1019

Should this be done better in both places, e.g. throwing an exception if the vector size does not fit `uchar`?

Reply:
There should indeed be a check that `polygon.size()<=std::numeric_limits<unsigned char>::max())`, skipping this property if that's the case.

Reply:
There is actually no "overflow" problem here as the size will simply be modulo `sizeof(uchar)`. Which means that less elements will be put.